### PR TITLE
Feature/product settings

### PR DESCRIPTION
**Pull request by nbradbury**
This is the feature branch for product settings, which includes:

- WP media gallery picker
- Product visibility
- Product menu order
- Product external url
- Product purchase note
- Product slug
- Product catalog visibility
- Product status
- Product short description

All code has been reviewed already.

Update release notes:

- [x] I have considered if this change warrants user-facing release notes and have added them to `RELEASE-NOTES.txt` if necessary.

### CODE REVIEW

File: WooCommerce/src/main/res/layout/fragment_product_catalog_visibility.xml [hunk 1]
```diff
@@ -0,0 +1,94 @@
...
+            <com.google.android.material.textview.MaterialTextView
+                android:textColor="@color/color_on_surface_high"
+                android:layout_width="match_parent"
+                android:layout_height="wrap_content"
+                android:padding="@dimen/major_100"
+                android:text="@string/product_visibility_headline" />
+        </com.woocommerce.android.widgets.WCElevatedLinearLayout>
+
+        <com.woocommerce.android.widgets.WCElevatedLinearLayout
+            style="Woo.TextView.Caption"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/fragment_product_catalog_visibility.xml:26]: np: Why is this style assigned to a layout?

Comment (nbradbury) [reply WooCommerce/src/main/res/layout/fragment_product_catalog_visibility.xml]: No good reason :) Fixed in 5c5090b.

File: WooCommerce/src/main/res/layout/fragment_product_status.xml [hunk 2]
```diff
@@ -0,0 +1,47 @@
+<?xml version="1.0" encoding="utf-8"?>
+<LinearLayout xmlns:android="http://schemas.android.com/apk/res/android"
+    xmlns:tools="http://schemas.android.com/tools"
+    android:id="@+id/radioGroup"
+    android:layout_width="match_parent"
+    android:layout_height="match_parent"
```
Comment (AmandaRiu) [inline WooCommerce/src/main/res/layout/fragment_product_status.xml:6]: Set background color to `?attr/colorSurface` so it's not gray in light mode.

**Review by AmandaRiu — APPROVED** [review]
@nbradbury Nice work on this feature. Since there have been a lot of design changes happening at the same time as this feature, I found some UI items that need to be tweaked and I've noted those in the comments. None of them are high priority or impact the functionality of this feature so I'm going to go ahead and approve this PR and open a new one with the UI updates. :shipit:

**Pull request merged**